src/jdk.jdi/share/classes/com/sun/tools/example/debug/tty/TTY.java

```diff
@@ -797,7 +799,7 @@ public TTY() throws Exception {
             this.handler = new EventHandler(this, true, trackVthreads);
         }
         try {
-            Charset charset = Charset.forName(System.getProperty("stdin.encoding"), Charset.defaultCharset());
+            Charset charset = Charset.forName(StaticProperty.stdinEncoding(), Charset.defaultCharset());
```

Comment:
This is the example jdb debugger, I assume okay to let it read the stdin.encoding property.